### PR TITLE
feat: use windows control overlay

### PR DESCRIPTION
**Pull request by mikelxk**
feat: use windows control overlay to use native control button on windows.
deps: upgrade electron 15
supersedes: #644 
Note:  electron 15 has completely deprecated the remote module so we need to go to the preload.js for embedding functions.
image
P.S: It seem there isn't any gap between theme buttons.

### CODE REVIEW

File: src/views/Home.vue [hunk 1]
```diff
@@ -774,7 +776,19 @@ const toggleMuteStatus = () => {
 const playFromPlaylist = (song) => {
   l1Player.playById(song.id);
 };
-
+onMounted(() => {
+  if ('windowControlsOverlay' in navigator) {
+    console.log(navigator.windowControlsOverlay.getBoundingClientRect())
```
Comment (listen1) [inline src/views/Home.vue:781]: please remove console.log before commit

**Comment by listen1** [comment]
this commit maybe needs more work to finish. 

*  isElectron function is not working properly, because electron v15 upgrade
* cookieGet/Set functions use remote module which is deprecated, so some migration work to be done.

File: src/provider/lowebutil.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ export function getParameterByName(name, url) {
 }
 
 export function isElectron() {
-  return window && window.process && window.process.platform;
+  return window.api?.platform;
```
Comment (listen1) [inline src/provider/lowebutil.js:14]: bug here: it will return platform name in electron environment, not a boolean value

**Comment by listen1** [comment]
bug detected: 
* in some situation when debug in electron, kuwo open playlist will load forever.

Kuwo server check `kw_token` cookie (which is used as csrf token) and this token will always remain same value even kuwo server response a new csrf token. So if some csrf is rejected by kuwo server, kuwo playlist will always loading.

So why this cookie is not refreshed? I think the reason is electron upgrade uses more strict policy for cross domain request ( for security consideration). If set-cookie header in response is not same site and samesite=None is not included in response cookie, the cookie will be discarded.

Maybe we can find some way to get cookie from response, but it seems hard for axios.

The only way so far is set a random csrf token (kuwo server will accept csrf not in blacklist, even it is not distributed by server). 

I think we'd better find a better way before we upgrade electron to v15.0

**Comment by mikelxk** [comment]
Downgrade to electron 14.
The console doesn't show a error any longer and a least some songs on kuwo front page works.
It seems that it's related with chromium change.
https://github.com/electron/electron/issues/24225

**Pull request merged**